### PR TITLE
refactor: unify dom lazy array classes

### PR DESCRIPTION
**Pull request by alandefreitas**
The DomCorpus uses multiple classes that represent lazy arrays. This commit simplifies the dom by unifying all of these array implementations. It's not a new design: it's only consolidating what already exists and removing duplicate code.

### CODE REVIEW

File: include/mrdocs/Metadata/Scope.hpp [hunk 1]
```diff
@@ -29,16 +29,29 @@ namespace mrdocs {
...
+    for other Info classes, such as NamespaceInfo,
+    ClassInfo, that represent scopes. This class
+    can also be used with composition, such as in
+    @ref Interface to represent different scopes of
+    the same class (such as member and static overloads).
+
 */
 struct ScopeInfo
 {
-	/** The members of this scope.
```
Comment (sdkrystian) [inline include/mrdocs/Metadata/Scope.hpp:35]: Unrelated change?

Comment (alandefreitas) [reply include/mrdocs/Metadata/Scope.hpp:35]: Mmm... Yes. A small indentation issue. I'll split the commits. :)

File: include/mrdocs/Metadata/Source.hpp [hunk 2]
```diff
@@ -36,6 +36,15 @@ MRDOCS_DECL
 std::string_view
 toString(FileKind kind);
 
+namespace dom {
+    template <>
+    struct ToValue<FileKind>
```
Comment (vinniefalco) [inline include/mrdocs/Metadata/Source.hpp:41]: why specialization instead of tag_invoke?

Comment (alandefreitas) [reply include/mrdocs/Metadata/Source.hpp]: done :)
